pilot/cmd/pilot-agent/request.go

```diff
@@ -50,3 +74,43 @@ var (
 func init() {
 	rootCmd.AddCommand(requestCmd)
 }
+
+// get admin port from Istio mesh configuration , This may be overridden by "ISTIO_PROXY_ADMIN_PORT" variable
+// , defaultEnvyConfigPath: /etc/istio/proxy/envoy-rev0.json or envoyAdminPort flags
+func getEnvoyHostAndAdminPort() (string, int32) {
+	var adminHost = "localhost"
+	var adminPort int32 = 15000
+
+	port, err := getPortFromProxyConfig()
+	if err == nil {
+		adminPort = port
+	}
+
+	if proxyAdminPortConfigEnv != 0 {
+		adminPort = int32(proxyAdminPortConfigEnv)
+	}
+
+	host, port, err := envoy.GetAdminHostAndPort(defaultEnvyConfigPath)
```

Comment:
Why do we still need this code if we are already reading from the above sources?

Reply:
Get the admin port from the mesh configuration, This may be overridden by "ISTIO_PROXY_ADMIN_PORT" environment variable or envory config   if it has been set.
so that the port can be obtained from multiple channels.

Reply:
We shouldn't allow so many ways to set the admin port, which is meaningless and makes confusion too

Reply:
Only get the admin port from the mesh configuration and override it from the `ISTIO_PROXY_ADMIN_PORT` environment variables if it has been set. it is okay?